ssl-impl/src/main/java/io/crate/protocols/postgres/ssl/SslConfiguration.java

```diff
@@ -0,0 +1,248 @@
+        } catch (final PrivilegedActionException e) {
+            throw (SSLException) e.getCause();
+        }
+
+        return sslContext;
+    }
+
+    @VisibleForTesting
+    static TrustStoreSettings loadTrustStore(Settings settings) throws KeyStoreException,
+                                                                       IOException,
+                                                                       NoSuchAlgorithmException,
+                                                                       CertificateException {
+        String trustStorePath = SslConfigSettings.SSL_TRUSTSTORE_FILEPATH.setting().get(settings);
+        checkStorePath(trustStorePath, StoreType.trust);
+        String trustStorePassword = SslConfigSettings.SSL_TRUSTSTORE_PASSWORD.setting().get(settings);
+
+        KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+
+        // load the store
+        trustStore.load(new FileInputStream(new File(trustStorePath)),
+                        trustStorePassword.isEmpty() ? null : trustStorePassword.toCharArray());
+
+        // initialize a trust manager factory with the trusted store
+        TrustManagerFactory trustFactory =
+            TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        trustFactory.init(trustStore);
+
+        // get the trust managers from the factory
+        TrustManager[] trustManagers = trustFactory.getTrustManagers();
+
+        return new TrustStoreSettings(trustStore, trustManagers);
+    }
+
+    @VisibleForTesting
+    static class TrustStoreSettings {
+
+        KeyStore trustStore;
+        TrustManager[] trustManagers;
+
+        TrustStoreSettings(KeyStore trustStore, TrustManager[] trustManagers) {
+            this.trustStore = trustStore;
+            this.trustManagers = trustManagers;
+        }
+    }
+
+    @VisibleForTesting
+    static KeyStoreSettings loadKeyStore(Settings settings) throws KeyStoreException,
+                                                                   IOException,
+                                                                   NoSuchAlgorithmException,
+                                                                   CertificateException,
+                                                                   UnrecoverableKeyException {
+        String keyStorePath = SslConfigSettings.SSL_KEYSTORE_FILEPATH.setting().get(settings);
+        checkStorePath(keyStorePath, StoreType.key);
+        String keyStorePassword = SslConfigSettings.SSL_KEYSTORE_PASSWORD.setting().get(settings);
+        String keyStoreKeyPassword = SslConfigSettings.SSL_KEYSTORE_KEY_PASSWORD.setting().get(settings);
+        if (keyStoreKeyPassword.isEmpty()) {
+            keyStoreKeyPassword = keyStorePassword;
+        }
+
+        // load the store
+        final KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        keyStore.load(new FileInputStream(new File(keyStorePath)),
```

Comment:
The FileInputStream isn't closed, so the file descriptor isn't freed (immediately)

It eventually will be freed if finalize is called on the instance, but it would probably be cleaner to wrap it into a try block